filesetup.c

```diff
@@ -833,12 +833,26 @@ static unsigned long long get_fs_free_counts(struct thread_data *td)
 uint64_t get_start_offset(struct thread_data *td, struct fio_file *f)
 {
 	struct thread_options *o = &td->o;
+	unsigned long long offset;
 
 	if (o->file_append && f->filetype == FIO_TYPE_FILE)
 		return f->real_file_size;
 
-	return td->o.start_offset +
-		td->subjob_number * td->o.offset_increment;
+	if (o->start_offset_percent > 0) {
+		// calculate the raw offset
+		offset = (f->real_file_size * o->start_offset_percent / 100) +
+			(td->subjob_number * td->o.offset_increment);
+
+		// block align the offset at the next available boundary
+		offset = (offset / o->ba[o->td_ddir] +
+			(offset % o->ba[o->td_ddir] != 0))  // ceiling(offset/blockalignment)
```

Comment:
@cbwest3 I don't think you do want it for fixed value offsets because people use those for purposely misaligning I/O (because you have exact granularity).